src/theory/arith/arith_rewriter.cpp

```diff
@@ -384,15 +384,40 @@ RewriteResponse ArithRewriter::postRewriteTranscendental(TNode t) {
         return RewriteResponse(REWRITE_DONE, NodeManager::currentNM()->mkConst(Rational(0)));
       }
     }else{
+      // get the factor of PI in the argument
       Node pi_factor;
       Node pi;
+      Node rem;
       if( t[0].getKind()==kind::PI ){
         pi_factor = NodeManager::currentNM()->mkConst(Rational(1));
         pi = t[0];
       }else if( t[0].getKind()==kind::MULT && t[0][0].isConst() && t[0][1].getKind()==kind::PI ){
         pi_factor = t[0][0];
         pi = t[0][1];
-      }
+      }else if( t[0].getKind()==kind::PLUS ){
+        for( unsigned j=0; j<t[0].getNumChildren(); j++ ){
```

Comment:
Let's split the logic here apart. Have a function that finds the index j and then logic to get pi, pi_factor and rem from j.

Reply:
Actually even better add a struct for the triple pi_factor, pi, and rem and have a function that returns values for the triple.